pkg/network/cache/client.go

```diff
@@ -75,3 +75,21 @@ func (c *ClientCache) Get(netAddr *network.Address) (client.Client, error) {
 
 	return cli, nil
 }
+
+// CloseAll closes underlying connections of all cached clients.
+//
+// Ignores closing errors.
+func (c *ClientCache) CloseAll() {
+	c.mu.RLock()
+
+	{
+		for _, cl := range c.clients {
+			con := cl.Conn()
+			if con != nil {
+				_ = con.Close()
+			}
```

Comment:
Don't you want to delete map records after connection close? I don't see how they can be reused after that.

Reply:
Lets for now just close: cache cleaning is not implemented in latest implementation, and the main point of the PR is to close clients.